thirdparties-extension/fr.opensagres.poi.xwpf.converter.core/src/main/java/fr/opensagres/poi/xwpf/converter/core/MasterPageManager.java

```diff
@@ -198,35 +203,76 @@ private void visitHeadersFooters( IXWPFMasterPage masterPage, CTSectPr sectPr )
-            if ( !firstHeaderFooter || ( firstHeaderFooter && !ignoreFirstHeaderFooter ) )
+            STHdrFtr type = reference.xgetType();
+            int typeValue = type == null ? STHdrFtr.INT_DEFAULT : type.enumValue().intValue();
+            if ( typeValue != STHdrFtr.INT_FIRST || titlePage )
             {
-                masterPage.setType( type.enumValue().intValue() );
-                documentHandler.visitHeaderRef( headerRef, sectPr, masterPage );
+                masterPage.setType( typeValue );
+                documentHandler.visitHeaderRef( reference, sectPr, masterPage );
+                masterPage.setType( typeValue );
+                headers.put( typeValue, masterPage.getHeader() );
             }
         }
-
-        for ( CTHdrFtrRef footerRef : footersRef )
+        for ( CTHdrFtrRef reference : sectPr.getFooterReferenceList() )
         {
-            STHdrFtr type = footerRef.xgetType();
-            firstHeaderFooter = ( type != null && type.enumValue() == STHdrFtr.FIRST );
-            if ( !firstHeaderFooter || ( firstHeaderFooter && !ignoreFirstHeaderFooter ) )
+            STHdrFtr type = reference.xgetType();
+            int typeValue = type == null ? STHdrFtr.INT_DEFAULT : type.enumValue().intValue();
+            if ( typeValue != STHdrFtr.INT_FIRST || titlePage )
             {
-                masterPage.setType( type.enumValue().intValue() );
-                documentHandler.visitFooterRef( footerRef, sectPr, masterPage );
+                masterPage.setType( typeValue );
+                documentHandler.visitFooterRef( reference, sectPr, masterPage );
+                masterPage.setType( typeValue );
+                footers.put( typeValue, masterPage.getFooter() );
             }
         }
+        if ( titlePage )
+        {
+            inheritHeader( masterPage, STHdrFtr.INT_FIRST, previousHeaders );
+            inheritFooter( masterPage, STHdrFtr.INT_FIRST, previousFooters );
+        }
+        if ( evenAndOddHeaders )
+        {
+            inheritHeader( masterPage, STHdrFtr.INT_EVEN, previousHeaders );
+            inheritFooter( masterPage, STHdrFtr.INT_EVEN, previousFooters );
+        }
+        inheritHeader( masterPage, STHdrFtr.INT_DEFAULT, previousHeaders );
+        inheritFooter( masterPage, STHdrFtr.INT_DEFAULT, previousFooters );
         masterPage.setType( STHdrFtr.INT_FIRST );
+    }
+
+    private void inheritHeader( IXWPFMasterPage<Object> aMasterPage, int aType, Map<Integer, Object> aPreviousHeaders )
+    {
+        if ( !headers.containsKey( aType ) )
+        {
+            Object header = aPreviousHeaders.get( aType );
+            if ( header != null )
+            {
+                aMasterPage.setType( aType );
+                aMasterPage.setHeader( header );
+                headers.put( aType, header );
+            }
+        }
+    }
 
+    private void inheritFooter( IXWPFMasterPage<Object> aMasterPage, int aType, Map<Integer, Object> aPreviousFooters )

```

Comment:
Please add comment by adding link to http://officeopenxml.com/WPSectionHeaderReference.php